rust/src/index/vector/graph/builder.rs

```diff
@@ -55,30 +80,54 @@ impl<V: Vertex> GraphBuilder<V> {
         &mut self.nodes[id].vertex
     }
 
-    pub fn neighbors(&self, id: usize) -> &[u32] {
-        self.nodes[id].neighbors.as_slice()
-    }
-
     pub fn neighbors_mut(&mut self, id: usize) -> &mut Vec<u32> {
         &mut self.nodes[id].neighbors
     }
 
-    pub fn add_edge(&mut self, from: usize, to: usize) {
-        self.nodes[from].neighbors.push(to as u32);
+    /// Set neighbors of a node.
+    pub fn set_neighbors(&mut self, id: usize, neighbors: impl Into<Vec<u32>>) {
+        self.nodes[id].neighbors = neighbors.into();
+    }
+
+    /// Add a neighbor to a specific vertex.
+    pub fn add_neighbor(&mut self, vertex: usize, neighbor: usize) {
+        self.nodes[vertex].neighbors.push(neighbor as u32);
     }
 }
 
-impl<V: Vertex> FromIterator<V> for GraphBuilder<V> {
-    fn from_iter<I: IntoIterator<Item = V>>(iter: I) -> Self {
-        let nodes: Vec<Node<V>> = iter
-            .into_iter()
-            .map(|v| Node {
-                vertex: v,
-                neighbors: vec![],
-            })
-            .collect();
+impl<V: Vertex + Clone> Graph for GraphBuilder<V> {
+    fn distance(&self, a: usize, b: usize) -> Result<f32> {
+        let vector_a = self.data.row(a).ok_or_else(|| {
+            Error::Index(format!(
+                "Attempt to access row {} in a matrix with {} rows",
```

Comment:
wht does this mean?

Reply:
Means the vector id is out of bound (of what are hold in the matrix view).  Suggestion for rewording?